File: 30DaysOfCode/17 - More Exceptions/mySolution.py
#Write your code here

# Calculator class with returns n raised to the power of p.
# If either value is negative, throw an exception to be caught by the provided
# code
class Calculator:
    def power(self, n, p):
        if n < 0 or p < 0:
            raise Exception("n and p should be non-negative")
        else:
            return n**p

# Provided code
myCalculator=Calculator()
T=int(input())
for i in range(T):
    n,p = map(int, input().split())
    try:
        ans=myCalculator.power(n,p)
        print(ans)
    except Exception as e:
        print(e)

File: 30DaysOfCode/19 - Interfaces _NOT PYTHON_/mySolution.java
// Provided code
import java.io.*;
import java.util.*;

interface AdvancedArithmetic{
  int divisorSum(int n);
}

//Write your code here
// Implement the divisiorSum function of the AdvancedArithmetic interface,
// ie. return the sum of all values which are even divisors of the input n
class Calculator implements AdvancedArithmetic {
  public int divisorSum(int n) {
    int sum = 0;
    // Check against all values up until n, starting at 1 to avoid mod by 0
    for (int i = 1; i <= n; i++) {
      // Only include in the sum if evenly divisible
      if (n % i == 0) {
          sum += i;
      }
    }

    return sum;
  }
}

// Provided code
class Solution {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int n = scan.nextInt();
    scan.close();

    AdvancedArithmetic myCalculator = new Calculator();
    int sum = myCalculator.divisorSum(n);
    System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
    System.out.println(sum);
  }
}
